core/src/main/java/io/questdb/ServerMain.java

```diff
@@ -292,6 +297,43 @@ public ServerMain(String[] args) throws Exception {
         }
     }
 
+    public static void setRssMemoryLimit(Log log, long rssMemoryLimit) {
+        try {
+            MBeanServer mBeanServer = ManagementFactory.getPlatformMBeanServer();
+
+            log.advisory().$("Java max memory: ").$size(Runtime.getRuntime().maxMemory()).$();
+
+            if (rssMemoryLimit == 0) {
+
+                Object freeMemSizeAttribute = mBeanServer.getAttribute(new ObjectName("java.lang", "type", "OperatingSystem"), "FreePhysicalMemorySize");
```

Comment:
I think this is tricky. 

Linux (and probably other OSs too) does aggressive disk caching & buffering. Basically, if it has no better use for memory then it will use it for disk buffering/caching. It's still better than letting it sit completely unused. Kernel can drop disk caches when more memory is needed. So I believe this under-reports *actually available* memory and can lead to unnecessary aggressive caps.

Reply:
Hmm, that's true. There is an option to override this though so user can set better limits.

Reply:
I would assume that by default we can use monopolize the box. That we can use all memory the box/vm/container has. (+respect cgroups cap, of course). I think that's the desirable state and it's also the case in containers. 

This means issues when questdb is collocated with other infra or embedded. but I would expect embedded users to be advanced and to set limits explicitly.

Reply:
I can change it to `TotalPhysicalMemorySize` and `TotalSwapSpaceSize`, what would be your formula?

We need to leave something for `mmaps` too.

Reply:
I'd keep the formula is it is. Except it would work with TotalPhysicalMemorySize instead of free memory. The formula already has a factor to keep something for the OS + various daemons. If you assume QuestDB can monopolize a box then the formula will work as it is.

Reply:
Ok, I changed it to Free Swap + Total RAM.